test/functional/test_framework/messages.py

```diff
@@ -293,17 +305,24 @@ def deserialize_v2(self, f):
         self.nServices = deser_compact_size(f)
 
         self.net = struct.unpack("B", f.read(1))[0]
-        assert self.net in (self.NET_IPV4, self.NET_I2P)
+        assert self.net in (self.NET_IPV4, self.NET_IPV6,
+                            self.NET_I2P, self.NET_CJDNS,
+                            self.NET_TORV2, self.NET_TORV3)
 
         address_length = deser_compact_size(f)
         assert address_length == self.ADDRV2_ADDRESS_LENGTH[self.net]
 
         addr_bytes = f.read(address_length)
         if self.net == self.NET_IPV4:
             self.ip = socket.inet_ntoa(addr_bytes)
-        else:
+        elif self.net == self.NET_IPV6:
+            self.ip = socket.inet_ntop(socket.AF_INET6, addr_bytes)
+        elif self.net == self.NET_I2P:
             self.ip = b32encode(addr_bytes)[0:-len(self.I2P_PAD)].decode("ascii").lower() + ".b32.i2p"
-
+        elif self.net == self.NET_CJDNS:
+           self.ip = ":".join(["{:02x}".format(b) for b in addr_bytes])
+        elif self.net in (self.NET_TORV2, self.NET_TORV3):
+            self.ip = b32encode(addr_bytes).decode('ascii') + '.onion'
```

Comment:
Is this correct for torv3? I had to add version and checksum in https://github.com/bitcoin/bitcoin/pull/27452

Reply:
You're correct, seems better your approach.